controllers/homework.js

```diff
@@ -162,169 +156,162 @@ router.post('/submit', getCreateHandler('submissions'));
-            var availableDateF = formattimepart(availableDate.getDate()) + "." + formattimepart(availableDate.getMonth() + 1) + "." + availableDate.getFullYear();
-            var availableTimeF = formattimepart(availableDate.getHours()) + ":" + formattimepart(availableDate.getMinutes());
-
-            var dueDateRaw = new Date(assignment.dueDate);
-            var dueDate = new Date(dueDateRaw.getTime() + (dueDateRaw.getTimezoneOffset() * 60000));
-            var dueDateF = formattimepart(dueDate.getDate()) + "." + formattimepart(dueDate.getMonth() + 1) + "." + dueDate.getFullYear();
-            var dueTimeF = formattimepart(dueDate.getHours()) + ":" + formattimepart(dueDate.getMinutes());
-
-            var now = new Date();
-            var remaining = (dueDate - now);
-            var remainingDays = Math.floor(remaining / (1000 * 60 * 60 * 24));
-            var remainingHours = Math.floor((    remaining % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
-            var remainingMinutes = Math.floor(((    remaining % (1000 * 60 * 60 * 24)) % (1000 * 60 * 60)) / (1000 * 60));
-            var dueColor;
-            var dueString;
-            if (remainingDays > 5 || remaining < 0) {
-                dueColor = "";
-                dueString = (dueDateF + " (" + dueTimeF + ")");
-            }
-            else if (remainingDays > 1) {
-                dueColor = "days";
-                dueString = "noch " + remainingDays + " Tage";
-            }
-            else if (remainingDays == 1) {
-                dueColor = "hours";
-                dueString = "noch " + remainingDays + " Tag " + remainingHours + ((remainingHours == 1) ? " Stunde" : " Stunden");
-            }
-            else if (remainingHours > 2) {
-                dueColor = "hours";
-                dueString = "noch " + remainingHours + " Stunden";
-            }
-            else if (remainingHours >= 1) {
-                dueColor = "minutes";
-                dueString = "noch " + remainingHours + ((remainingHours == 1) ? " Stunde " : " Stunden ") + remainingMinutes + ((remainingMinutes == 1) ? " Minute" : " Minuten");
-            }
-            else {
-                dueColor = "minutes";
-                dueString = "noch " + remainingMinutes + ((remainingMinutes == 1) ? " Minute" : " Minuten");
-            }
-
-
-            assignment.dueColor = dueColor;
-            if((assignment.teacherId == res.locals.currentUser._id) && (remainingDays > 5 || remaining < 0)) {
-                assignment.fromdate = availableDateF + " (" + availableTimeF + ")";
-                assignment.todate = dueDateF + " (" + dueTimeF + ")";
+            assignment.privateclass = assignment.private ? "private" : ""; // Symbol für Private Hausaufgabe anzeigen?
+
+            // Anzeigetext + Farbe für verbleibende Zeit
+            var availableDateArray = splitDate(assignment.availableDate);
+            var dueDateArray = splitDate(assignment.dueDate);
```

Comment:
Please use const/let unless there is a reason to use var. Especially in the controller as node supports const/let natively.